src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -817,6 +817,11 @@ private static bool IsEndTag(Block tagBlock)
             return relevantSymbol.Type == HtmlSymbolType.ForwardSlash;
         }
 
+        private static bool IsCommentTag(Span span)
+        {
+            return span.Content.StartsWith("<!--");
```

Comment:
I think we should verify `Span.Kind` here instead. We have a `Comment` kind.

Reply:
HTML Comments are treated as `Markup`, not `Comment`